src/overlays/actors/ovl_En_Stop_heishi/z_en_stop_heishi.c

```diff
@@ -32,45 +46,529 @@ ActorInit En_Stop_heishi_InitVars = {
+        if (this->unk_27E != 0) {
+            this->actor.textId = this->unk_27E;
+            func_80151938(play, this->actor.textId);
+            EnStopHeishi_ChangeAnim(this, SOLDIER_ANIM_STAND_HAND_ON_HIP);
+            gSaveContext.save.weekEventReg[0xC] |= 0x20;
+            this->unk_265 = 1;
+            this->unk_27E = 0;
+        } else {
+            func_801477B4(play);
+            func_80AE7E9C(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/func_80AE7E9C.s")
+void EnStopheishi_Update(Actor* thisx, PlayState* play) {
+    EnStopheishi* this = (EnStopheishi*)thisx;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/func_80AE7F34.s")
+    if (this->timer != 0) {
+        this->timer = this->timer - 1;
+    }
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    Math_SmoothStepToS(&this->headXRot, this->headXRotTarget, 1, 0x7D0, 0);
+    Math_SmoothStepToS(&this->headZRot, this->pitchToPlayer, 1, 0x7D0, 0);
+    this->actionFunc(this, play);
+    Actor_MoveWithGravity(&this->actor);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 20.0f, 20.0f, 50.0f, 0x1D);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.uncullZoneForward = 500.0f;
+    Math_Vec3f_Copy(&this->actor.focus.pos, &this->headWorldPos);
+    Math_Vec3s_Copy(&this->actor.focus.rot, &this->actor.world.rot);
+    if (!this->disableCollider) {
+        Collider_UpdateCylinder(&this->actor, &this->collider);
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/func_80AE854C.s")
+s32 EnStopheishi_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnStopheishi* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/func_80AE85C4.s")
+    if (limbIndex == SOLDIER_LIMB_HEAD) {
+        rot->x += this->headXRot;
+        rot->z += this->headZRot;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/EnStopheishi_Update.s")
+void EnStopheishi_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnStopheishi* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/func_80AE87A4.s")
+    if (limbIndex == SOLDIER_LIMB_HEAD) {
+        Matrix_MultVec3f(&gZeroVec3f, &this->headWorldPos);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/func_80AE87EC.s")
+void EnStopheishi_Draw(Actor* thisx, PlayState* play) {
+    EnStopheishi* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stop_heishi/EnStopheishi_Draw.s")
+    func_8012C28C(play->state.gfxCtx);
+    SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, (s32)this->skelAnime.dListCount,
```

Comment:
Is this cast required?
```suggestion
    SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
```